### ISSUE TITLE
linarize text

### ISSUE THREAD

**Issue by jkcapoeira** [qcz/vscode-text-power-tools#23]
Hi,
is there also an option to linarize a block of text. The idea is that i can select a block of code, copy it in a new tab, and then linarize it so that all the returns are removed, and it will become 1 line. After that i can format that as json or xml. (like in the notepad++ the XML Tools plugin with the function linarize XML)

**Comment by qcz** [comment]
If I understand it correctly, you would like the the following:

this
```
aa
bb
cc
```
to
```
aabbcc
```

and this

```ts
export function compareNumbers(a: number , b: number) {
	return a < b ? -1 : a > b ? 1 : 0;
}
```
to
```ts
export function compareNumbers(a: number , b: number) {	return a < b ? -1 : a > b ? 1 : 0;}
```

?

What about leading/trailing whitespace?

**Comment by jkcapoeira** [comment]
yes, correct. I did not think about leading and trailing spaces, but that
would be good also. I have those files where the left column is the hex
presentation and in the right is the ascii text. so I select only the right
column and want to format it as json or xml or whatever the content is.
thanks for the quick reply ;

**Comment by qcz** [comment]
_Remove newlines_ and _Trim whitespace and remove newlines_ commands are available in the new 1.29.0 version.

**Comment by jkcapoeira** [comment]
Great,thanks

**Comment by franzbischoff** [comment]
Hello,
I was looking for another feature but could not find. It is almost like this, but adding a space and removing >1 spaces. Example:

```
Sed ut perspiciatis unde omnis  iste natus error sit voluptatem
accusantium doloremque    laudantium, totam rem aperiam, eaque
ipsa quae ab illo inventore    veritatis et quasi architecto
```

would become:

```
Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto
```

Is it possible already?

Thanks.

**Comment by qcz** [comment]
It is possible in the new version (1.32.0).
Use the _Replace new lines and whitespace with a single space_ command.

**Comment by qcz** [comment]
@franzbischoff It is now possible.
Use the _Replace new lines and whitespace with a single space_ command in the newly published version (1.32.0).
